Recipes/Recipes/View Controllers/MainViewController.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  MainViewController.swift
+//  Recipes
+//
+//  Created by Jesse Ruiz on 9/30/19.
+//  Copyright © 2019 Lambda Inc. All rights reserved.
+//
+
+import UIKit
+
+class MainViewController: UIViewController {
+    
+    // MARK: - Properties
+    let networkClient = RecipesNetworkClient()
+    var allRecipes: [Recipe] = [] {
+        didSet {
+            filterRecipes()
+        }
+    }
+    var recipesTableViewController: RecipesTableViewController? {
+        didSet {
+            recipesTableViewController?.recipes = filteredRecipes
+        }
+    }
+    var filteredRecipes: [Recipe] = [] {
+        didSet {
+            recipesTableViewController?.recipes = filteredRecipes
+        }
+    }
+    
+    // MARK: - Outlets
+    @IBOutlet weak var textField: UITextField!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        networkClient.fetchRecipes { (recipes, error) in
+            if let anError = error {
+                NSLog("There was an error: \(anError)")
+            } else {
+                guard let recipe = recipes else { return }
+                self.allRecipes = recipe
+            }
+        }
+    }
+    
+    // MARK: - Actions
+    @IBAction func actionTextField(_ sender: UITextField) {
+        resignFirstResponder()
+        filterRecipes()
+    }
+    
+    // MARK: - Methods
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "displayTableView" {
+            recipesTableViewController = segue.destination as? RecipesTableViewController
+        }
+    }
+    
+    func filterRecipes() {
+        guard let text = textField.text else { return }
+        if text.isEmpty {
+            filteredRecipes = allRecipes
+        } else {
+            filteredRecipes = allRecipes.filter({ $0.name == text || $0.instructions == text})
+        }
+    }
+    
```

Comment:
What you have from line 59 to 66 will only return the values if you type the name completely how it is stored.  If you changed the conditions to check if the name contains(text) you will have different results.  try changing this function to 
```
 func filterRecipes() {
           guard let searchTerm = textField.text, !searchTerm.isEmpty else {
               filteredRecipes = allRecipes
               return
           }
           filteredRecipes = allRecipes.filter {
               $0.name.contains(searchTerm) || $0.instructions.contains(searchTerm)
           }
       }
```